lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -704,6 +712,17 @@ def get_docker_environment(env, env_files):
     return sorted(env_list)
 
 
+def get_nanoseconds_from_raw_option(value):
+    if value is None:
+        return None
+    elif isinstance(value, int):
+        return value
+    elif isinstance(value, string_types):
+        return convert_duration_to_nanosecond(value)
```

Comment:
It can also happen that the argument has been passed in as a string before; Ansible would have converted it to an `int` if it was trivially convertable. `convert_duration_to_nanosecond` will fail on such strings.

You should probably add something like
```suggestion
        try:
            return int(value)
        except ValueError as e:
            return convert_duration_to_nanosecond(value)
```